### PR TITLE
Add help with order of keys

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add help with order of keys

## Motivation and Context
This is the new change to capture the right order of keys

## How to test this PR?
mc side changes are not done yet

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [x] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#ae0b7cec-054c-4552-8dd0-d491f0a2c84d) |
<a name="ae0b7cec-054c-4552-8dd0-d491f0a2c84d"></a>
### 8535-3d4ff14/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:32077
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cf8b9ec70948:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 23 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 35 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 22 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 21 seconds
{
  "args": {},
  "duration": 110990,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8535-3d4ff14

**Comment by poornas** [comment]
@harshavardhana, minor bug in output here - should have shown
```
compression extensions=".txt" comment="my fancy config for compression" 
```
```
➜  mc git:(099226d1) mc admin config get myminio compression 
compression extensions=".txt comment my fancy config for compression" 
```
oops - it was just that a bad extension was allowed to be set - pls ignore.
```mc admin config set myminio compression state=on extensions=".txt" comment "my fancy config for compression"```

**Comment by harshavardhana** [comment]
@poornas was this with some existing data? if yes then please purge and start fresh to reproduce this issue.

**Comment by harshavardhana** [comment]
```
~ mc admin config set myminio compression extensions=".txt" comment="my fancy config for compression" 
Setting new key has been successful.
Please restart your server with `mc admin service restart myminio`.
```

```
~ mc admin config get myminio compression
compression extensions=.txt comment="my fancy config for compression"
```

Make sure you have updated your `mc` as well @poornas

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#67fa4ce5-507b-4cd7-a989-fb09fd41620b) |
| mint-large-bucket.sh | :x: [more...](#4a4dc889-8d79-4751-9e42-8792ff280a80) |
<a name="4a4dc889-8d79-4751-9e42-8792ff280a80"></a>
### 8535-c4093f8/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31753
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0dc2d93b9930:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 47 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 20 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 25 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 35 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 22 seconds
{
  "args": {},
  "duration": 68374,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="67fa4ce5-507b-4cd7-a989-fb09fd41620b"></a>
### 8535-c4093f8/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30142
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fa8c8633b61c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 1 minutes and 0 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-dotnet tests ... done in 25 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 7 seconds
(10/14) Running minio-java tests ... done in 22 seconds
(11/14) Running minio-js tests ... FAILED in 11 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 9,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:83:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:153:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:77:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:607:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:616:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:629:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:567:22) at IncomingMessage.onend (_stream_readable.js:514:10) at endReadableNT (_stream_readable.js:978:12) at _combinedTickCallback (internal/process/next_tick.js:80:11) at process._tickCallback (internal/process/next_tick.js:104:9)"
}

Executed 10 out of 14 tests successfully.
```
Error: No such image: minio/minio:8535-c4093f8

File: cmd/config-current.go [hunk 1]
```diff
@@ -261,7 +260,111 @@ func lookupConfigs(s config.Config) (err error) {
...
+	},
+	config.HelpKV{
+		Key:             config.NotifyWebhookSubSys,
+		Description:     "Configure to publish events to Webhook target",
+		MultipleTargets: true,
+	},
+	config.HelpKV{
+		Key:             config.NotifyESSubSys,
+		Description:     "Configure to publish events to Elasticsearch target",
+		MultipleTargets: true,
```
Comment (poornas) [inline cmd/config-current.go:364]: would be good to move ES up to alphabetize the listing of notification targets.

File: cmd/config/notify/help.go [hunk 2]
```diff
@@ -23,141 +23,650 @@ import (
...
+		},
+		config.HelpKV{
+			Key:         target.AmqpMandatory,
+			Description: "Set this to 'on' for server to return an unroutable message with a Return method. If this flag is 'off', the server silently drops the message",
+			Optional:    true,
+			Type:        "on|off",
+		},
+		config.HelpKV{
+			Key:         target.AmqpDurable,
+			Description: "Set this to 'on' for queue to surive broker restarts",
```
Comment (poornas) [inline cmd/config/notify/help.go]:
```suggestion
 			Description: "Set this to 'on' for queue to survive broker restarts",
```

File: cmd/config/notify/help.go [hunk 3]
```diff
@@ -23,141 +23,650 @@ import (
...
+			Key:         target.KafkaTopic,
+			Description: "The Kafka topic for a given message",
+			Optional:    true,
+			Type:        "string",
+		},
+		config.HelpKV{
+			Key:         target.KafkaSASLUsername,
+			Description: "Username for SASL/PLAIN  or SASL/SCRAM authentication",
+			Optional:    true,
+			Type:        " string",
```
Comment (poornas) [inline cmd/config/notify/help.go]:
```suggestion
 			Type:        "string",
```

File: cmd/config/notify/help.go [hunk 4]
```diff
@@ -23,141 +23,650 @@ import (
...
+			Type:        "address",
+		},
+		config.HelpKV{
+			Key:         target.RedisFormat,
+			Description: "Specify how data is populated, a hash is used in case of `namespace` format and a list in case of `access` format, defaults to 'namespace'",
+			Type:        "namespace|access",
+		},
+		config.HelpKV{
+			Key:         target.RedisKey,
+			Description: "The name of the redis key under which events are stored",
```
Comment (poornas) [inline cmd/config/notify/help.go]:
```suggestion
 			Description: "The name of the Redis key under which events are stored",
```

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/config-current.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/config-current.go:364]: There is no alphabetic sorting, the order is manual @poornas I will move elasticsearch above and move webhook higher than all targets.

**Review by poornas — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#b3963a25-9260-4327-86fa-4e20e3d77489) |
<a name="b3963a25-9260-4327-86fa-4e20e3d77489"></a>
### 8535-25924e5/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30125
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 397eee86062b:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 4 seconds
(3/14) Running aws-sdk-php tests ... done in 57 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 40 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 23 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 8 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 31 seconds
{
  "args": {},
  "duration": 85687,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8535-25924e5

**Pull request merged**